internal/app/email/mailservice.go

```diff
@@ -0,0 +1,150 @@
+package email
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/TheFlies/ofriends/internal/app/customer"
+	"github.com/TheFlies/ofriends/internal/app/cusvisitassoc"
+	"github.com/TheFlies/ofriends/internal/app/types"
+	"github.com/TheFlies/ofriends/internal/app/visit"
+	envconfig "github.com/TheFlies/ofriends/internal/pkg/config/env"
+	"github.com/TheFlies/ofriends/internal/pkg/glog"
+	"github.com/TheFlies/ofriends/internal/pkg/mail"
+)
+
+type (
+	SendMailService struct {
+		VisitRepo    visit.Repository
+		CustomerRepo customer.Repository
+		CusAssVis    cusvisitassoc.Repository
+		logger       glog.Logger
+	}
+
+	TimeConf struct {
+		/*
+			the service will send email to notify internal communication the visit is coming (day)
+			default is 7days = one week before visit come
+		*/
+		SendingCondition string `envconfig:"SENDING_CONDITION" default:"7"`
+		TimeZoneLocation string `envconfig:"TIME_ZONE_LOCATION" default:"Asia/Bangkok"`
+	}
+)
+
+const OFRIENDS_DATE_FORMATE = "15:04 Jan 02"
+
+func NewSendMailService(visitRepo visit.Repository, customerRepo customer.Repository, assoc cusvisitassoc.Repository, l glog.Logger) *SendMailService {
+	return &SendMailService{
+		VisitRepo:    visitRepo,
+		CustomerRepo: customerRepo,
+		CusAssVis:    assoc,
+		logger:       l,
+	}
+}
+func (m *SendMailService) Send(ctx context.Context, now time.Time) error {
+	var conf TimeConf
+	envconfig.Load(&conf)
+
+	sendingCondition, err := strconv.Atoi(conf.SendingCondition)
```

Comment:
So why not just define `conf.SendingCondition` as `int`?

Reply:
Because i want to validate  environment variable. i can't you the string and after that I convert this variable and i can c check accuracy of variable, but I have forgotten in the `envconfig` lib can validate the variable. I will fix it